src/deploymentTemplate.ts

```diff
@@ -0,0 +1,140 @@
+                // XXX - to be implemented
+            } else if (selection == items[2]) {
+                // XXX - to be implemented
+            }
+            
+        });
+    }
+
+    public selectQuickstartTemplate() {
+
+        // get list of directories from here:
+        // https://api.github.com/repos/Azure/azure-quickstart-templates/contents/
+
+        var http = require('https');
+        let __this = this;
+
+            http.get({
+                host: 'api.github.com',
+                path: '/repos/Azure/azure-quickstart-templates/contents/',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+            }, function(response) {
+                // Continuously update stream with data
+                var body = '';
+                response.on('data', function(d) {
+                    body += d;
+                });
+                response.on('end', function() {
+        
+                    // Data reception is done, do whatever with it!
+                    var parsed = JSON.parse(body);
+                    let items: vscode.QuickPickItem[] = [];
+            
+                    for (var i in parsed)
+                    {
+                        if (parsed[i].type == "dir") {
+                            items.push({ 
+                                label: parsed[i].name, 
+                                description: parsed[i].name });
+                        }
+
+                    }
+                    vscode.window.showQuickPick(items).then(selection => {
+                        // the user canceled the selection
+                        if (!selection) {
+                          return;
+                        }        
+                        
+                        __this.retrieveTemplate(selection.label);
+                    });
+                });
+            });
+        
+    }
+
+    public retrieveTemplate(templateName: string) {
+        var http = require('https');
+        let __this = this;
+
+            http.get({
+                host: 'raw.githubusercontent.com',
+                path: '/Azure/azure-quickstart-templates/master/' + templateName + '/azuredeploy.json',
+                headers: { 'User-Agent': 'VSC Ansible Extension'}
+            }, function(response) {
+                // Continuously update stream with data
+                var body = '';
+                response.on('data', function(d) {
+                    body += d;
+                });
+                response.on('end', function() {
+        
+                    // Data reception is done, do whatever with it!
+                    var parsed = JSON.parse(body);
+
+                    // XXX - get parameters
+                    // XXX - create url
+
+                    // XXX - create playbook
+                    __this.createPlaybookFromTemplate("https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/" + templateName + '/azuredeploy.json',
```

Comment:
add those constants to Constants.ts?

Reply:
just spoke to @yaweiw , we'll have configurable template repo URL

Reply:
User should be able to use their collection of ARM templates. 
* Deploy from Azure Quick Start ARM Templates
* Deploy from User ARM Templates

Also, i think repo and local/network path should also be supported.

Reply:
perhaps we should have a list of GitHub repos on our configuration file, so user could add a few urls, then we would display choice:
- Azure Quickstart Templates
- User xxx xxx templates
- Enter template location
if user selects "Enter template location", there would be just input box, where user would have to enter either URL or local path to the template.